db/versions/0072.yml

```diff
@@ -0,0 +1,176 @@
+version: 72
+description: add last_date_active to queue_workers
+migrationScript: |
+  begin
+    alter table queue_workers add column last_date_active timestamp with time zone not null;
+  end
+downgradeScript: |
+  begin
+    alter table queue_workers drop column last_date_active;
+  end
+methods:
+  quarantine_queue_worker:
+    deprecated: true
+  get_queue_worker_tqid:
+    deprecated: true
+  get_queue_workers_tqid:
+    deprecated: true
+  queue_worker_seen:
+    description: |-
+      Recognize that a worker has been seen by the queue, creating it if necessary.  This is called
+      when workers claim or re-claim work.  The expiration time is not allowed to move backward.
+
+      This function always writes to the DB, so calls should be suitably rate-limited at the
+      client side.
+    deprecated: true
```

Comment:
as I understand it, if we deprecate a function, it should not be changed, so all other props are not needed here

Reply:
So since we guarantee all DB functions must work for 2 major TC versions, this needed to be updated. Check out the points under the first checkbox [here](https://github.com/taskcluster/taskcluster/tree/main/db#checklist).

Reply:
I was thinking more of [this](https://github.com/taskcluster/taskcluster/tree/main/db#stored-functions). Basically signature changes (new return column), so name should also be changed as I understand this

Reply:
Ah ok I see. So not having a re-implementation of this DB func fails tests and would fail if anyone uses it because the new column has a not-null constraint.

So for example, if I changed this to just `deprecated: true`, the output of `yarn workspace taskcluster-db test` is:
```bash
  463 passing (1m)
  1 failing

  1) db/test/fns/queue_test.js
       queue_workers
         queue_worker_seen (deprecated):
     error: null value in column "last_date_active" violates not-null constraint
DETAIL: Failing row contains (wg, wi, [], 2012-04-21 13:21:16.684619+00, 2022-04-22 13:21:16.684+00, 2022-04-21 13:21:16.684619+00, prov/wt, null).
WHERE: SQL statement "insert
    into queue_workers (task_queue_id, worker_group, worker_id, quarantine_until, expires, first_claim, recent_tasks)
    values (
      task_queue_id_in,
      worker_group_in,
      worker_id_in,
      now() - interval '10 years',
      expires_in,
      now(),
      jsonb_build_array()
    )
    on conflict (task_queue_id, worker_group, worker_id) do update
    set
      expires = greatest(coalesce(expires_in, queue_workers.expires), queue_workers.expires)
    where
      queue_workers.task_queue_id = task_queue_id_in and
      queue_workers.worker_group = worker_group_in and
      queue_workers.worker_id = worker_id_in"
PL/pgSQL function queue_worker_seen(text,text,text,timestamp with time zone) line 2 at SQL statement
CODE: 23502
```

Reply:
Oh, that probably means that upgrade script will also fail, since all of the existing records do not have default value, and field itself is marked as `not null`